File: android/src/main/kotlin/com/mikoloy/device_trust/DeviceTrustPlugin.kt
package com.mikoloy.device_trust

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

/** DeviceTrustPlugin */
class DeviceTrustPlugin : FlutterPlugin, MethodChannel.MethodCallHandler {
  private lateinit var channel: MethodChannel
  private lateinit var appContext: Context

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    appContext = binding.applicationContext
    channel = MethodChannel(binding.binaryMessenger, "device_trust")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
    when (call.method) {
      "getDeviceTrustReport" -> {
        try {
          val report = DeviceTrust.buildReport(appContext)
          result.success(report.toWirePayload())
        } catch (e: Exception) {
          result.error("DEVICE_TRUST_ERROR", e.message, null)
        }
      }
      else -> result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}

File: ios/device_trust/Sources/device_trust/DeviceTrustPlugin.swift
import Flutter
import UIKit

public class DeviceTrustPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "device_trust", binaryMessenger: registrar.messenger())
    let instance = DeviceTrustPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getDeviceTrustReport":
      // Call DeviceTrust.buildReport() defined in DeviceTrust.swift
      let report = DeviceTrust.buildReport()
      result(report.toMap())
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}